management/server/nameserver.go

```diff
@@ -364,9 +371,8 @@ func validateDomainInput(primary bool, domains []string) error {
 			" you should set either primary or domain")
 	}
 	for _, domain := range domains {
-		_, valid := dns.IsDomainName(domain)
```

Comment:
why create a new function instead of improve the dns.IsDomainName?

Reply:
`dns.IsDomainName` is not enough to cover all valid domain cases. I moved to new function that includes regex for matching other requirements